pkg/asset/installconfig/alibabacloud/validation.go

```diff
@@ -190,3 +192,23 @@ func validateClusterName(client *Client, ic *types.InstallConfig) field.ErrorLis
 	}
 	return allErrs
 }
+
+func validateMetadataServerIPNotInMachineCIDR(n *types.Networking) field.ErrorList {
+	allErrs := field.ErrorList{}
+	fldPath := field.NewPath("networking").Child("machineNetwork")
+	matedataServerIP := "100.100.100.200"
+	if err := validateIPNotinMachineCIDR(matedataServerIP, n); err != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath, matedataServerIP, err.Error()))
+	}
+	return allErrs
+
+}
+
+func validateIPNotinMachineCIDR(ip string, n *types.Networking) error {
+	for _, network := range n.MachineNetwork {
+		if network.CIDR.Contains(net.ParseIP(ip)) {
+			return fmt.Errorf("the IP must not be in one of the machine networks")
```

Comment:
"the IP must not be in one of the machine networks" -> "the <network>  contains 100.100.100.200 which is reserved for the metadata service"